src/App.js

```diff
@@ -23,49 +23,108 @@ const generateSquares = () => {
   }
 
   return squares;
-}
+};
 
 const App = () => {
-
+  console.log(generateSquares());
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
 
   // Wave 2
-  // You will need to create a method to change the square 
+  // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
-
+  const onClickCallback = (clickedSquare) => {
+    const newSquares = [...squares];
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 3; j++) {
+        if (checkForWinner() === null) {
+          if (newSquares[i][j].id === clickedSquare.id) {
+            if (newSquares[i][j].value === '') {
+              newSquares[i][j].value = currentPlayer;
+              if (currentPlayer === PLAYER_1) {
+                setCurrentPlayer(PLAYER_2);
+              } else {
+                setCurrentPlayer(PLAYER_1);
+              }
+            }
+          }
+        }
+      }
+    }
```

Comment:
We can reduce how nested this code block is by doing the following: 

1. Checking if there's a winner before starting the for loop, preventing unneeded iteration. 
2. Combining the conditionals to check if the `id` match and contain an empty `value`. 
3. Using a ternary to set the current player


```suggestion
  if (!checkForWinner()) {
    for (let i = 0; i < 3; i++) {
      for (let j = 0; j < 3; j++) {
          if (newSquares[i][j].id === clickedSquare.id && newSquares[i][j].value === '') {
              newSquares[i][j].value = currentPlayer;
             setCurrentPlayer( currentPlayer === PLAYER_1 ? PLAYER_2 : PLAYER_1) 
          }
        }
      }
    }
```